lib/spack/spack/detection/path.py

```diff
@@ -87,7 +87,9 @@ def libraries_in_ld_library_path(path_hints=None):
         path_hints (list): list of paths to be searched. If None the list will be
             constructed based on the LD_LIBRARY_PATH environment variable.
     """
-    path_hints = path_hints or spack.util.environment.get_path('LD_LIBRARY_PATH')
+    path_hints = path_hints or \
+        spack.util.environment.get_path('LD_LIBRARY_PATH') or \
+        spack.util.environment.get_path('DYLD_LIBRARY_PATH')
```

Comment:
Why do we only search `LD_LIBRARY_PATH` and not `LIBRARY_PATH`? Also, you should add `DYLD_FALLBACK_LIBRARY_PATH`, this is what Spack uses.

Reply:
Done.